esphome/components/thermostat/climate.py

```diff
@@ -431,7 +431,8 @@ async def to_code(config):
 
     heat_cool_mode_available = CONF_HEAT_ACTION in config and CONF_COOL_ACTION in config
     two_points_available = CONF_HEAT_ACTION in config and (
-        CONF_COOL_ACTION in config or CONF_FAN_ONLY_ACTION in config
+        CONF_COOL_ACTION in config
+        or (CONF_FAN_ONLY_ACTION in config and config[CONF_FAN_ONLY_COOLING] is True)
```

Comment:
Do you need to explicitely check for `True` or is truthy also ok?

Also checking truthy is faster than an `in` check, so if you turn them around, it is teeny tiny faster.

```suggestion
        or (config[CONF_FAN_ONLY_COOLING] and CONF_FAN_ONLY_ACTION in config)
```

Reply:
Ah yeah, great catch -- and good for me to learn that may be a tiny bit faster. Let's do it!

Reply:
> Also checking truthy is faster than an `in` check, so if you turn them around, it is teeny tiny faster.

Actually, you can't do that (in general), because the truthy check throws if the item isn't present, so you need to check existence first. In this case that won't happen though, as the config option has a default value, so it'll always be present (and the existence check is thus superfluous.